app/services/bulk_tagging/document_type_tagger.rb

```diff
@@ -0,0 +1,36 @@
+module BulkTagging
+  class DocumentTypeTagger
+    def self.call(taxon_base_path:, document_type:)
+      new(taxon_base_path: taxon_base_path, document_type: document_type).call
+    end
+
+    def initialize(taxon_base_path:, document_type:)
+      @taxon_base_path = taxon_base_path
+      @document_type = document_type
+    end
+
+    def call
+      taxon_content_id = Services.publishing_api.lookup_content_id(base_path: @taxon_base_path)
+      raise StandardError, "Cannot find taxon with base path #{@taxon_base_path}" if taxon_content_id.nil?
+
+      Services.publishing_api.get_content_items_enum(document_type: @document_type, fields: ['content_id']).lazy.map do |document|
+        begin
+          content_id = document.fetch('content_id')
+          add_taxon_link(content_id, taxon_content_id)
+          { status: 'success', message: 'success', content_id: content_id }
+        rescue StandardError => ex
+          { status: 'error', message: ex.message, content_id: content_id }
+        end
+      end
+    end
+
+  private
+
+    def add_taxon_link(content_id, taxon_content_id)
+      response_hash = Services.publishing_api.get_links(content_id).to_h
+      version = response_hash['version']
+      new_links = response_hash['links'].merge('taxons' => [taxon_content_id])
```

Comment:
I think this will have the same behaviour as before, to really add the `taxon_content_id` to the existing links you will need to combine the list of content ids.